### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by shihabrafio**
In this PR, I have done
- Add Redux Toolkit 
- Create a directory that will contain all your Redux logic
- Configure a Redux store
- Define a slice of state for books that:
- Includes an array of books (initial state: empty array)
  - Includes a reducer that adds a book
  - Includes a reducer that removes a book
- Define a slice of state for categories

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+const addbook = 'bookStore/books/addbook';
+const removebook = 'bookStore/books/removebook';
+
+const initialbooks = {
+  books: [],
+};
+
```
Comment (t-yanick) [inline src/redux/books/books.js]: - [ ] Great job working on this milestone. As per project requirements, you are supposed to make use of/define `slice` for the redux state. At the moment, you have not used slices. Even the import of `createSlice ` from `"@reduxjs/toolkit"` you don't have it.
You can have something similar to below snippet before the reducers for adding and removing books
```javascript
import { createSlice } from "@reduxjs/toolkit"

export const booksSlice = createSlice({
  name: "books",
  initialState: {
    books: [],
  },
    .
    .
    .
```
image

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  type: addbook,
+  id: 1,
+  title: 'Harry Potter',
+});
+
+export const removeBooks = (id) => ({
+  type: removebook,
+  id,
+});
+
```
Comment (t-yanick) [inline src/redux/books/books.js]: - [ OPTIONAL] 
For more clarity you can rename this file to` booksSlice.js`

File: src/redux/catergories/catergories.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const checkcategory = 'bookstore/categories/checkcategory';
+
+const initialcategories = {
+  category: [],
+};
+
```
Comment (t-yanick) [inline src/redux/catergories/catergories.js]: - [ ] Great job working on this milestone. As per project requirements, you are supposed to make use of/define `slice` for the redux state. At the moment, you have not used slices. Even the import of `createSlice ` from `"@reduxjs/toolkit"` you don't have it.
You can have something similar to below snippet before the reducer for checking status
```javascript
import { createSlice } from "@reduxjs/toolkit"

export const categoriesSlice = createSlice({
  name: "categories",
  initialState: {
    categories: [],
  },
    .
    .
    .
```
image

File: src/redux/catergories/catergories.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+// Create Reducer
+export const checkCategory = () => ({
+  type: checkcategory,
+});
+
```
Comment (t-yanick) [inline src/redux/catergories/catergories.js:15]: - [ OPTIONAL] 
For more clarity you can rename this file to` categoriesSlice.js`

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @shihabrafio , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Required packages present in package.json
:heavy_check_mark: File structure follows ducks pattern
:heavy_check_mark: Correct Gitflow used
:heavy_check_mark: No linters error

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shisui6 — APPROVED** [review]
Hi @shihabrafio, 

# Approved 🪄
Your project is complete! There is nothing else to say other than... it's time to merge it 💎
Congratulations! 🎉 

## Highlights 🎯
- Added actions and reducers ✅


## Optional suggestions 🪶

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will really improve your application._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**